File: Plotting_Scripts/Figure6/Figure6.jl
using GMT

# change the scale of the default cooridinates
function preset()
    xmin = -6400
    xmax = 6400
    ymin = 0
    ymax = 6000
    scale = 1000
    xmin = xmin / scale
    xmax = xmax / scale
    ymin = ymin / scale
    ymax = ymax / scale

    println("$xmin $xmax $ymin $ymax")
    range_xy = "$xmin/$xmax/$ymin/$ymax"
    return range_xy
end

range_xy = preset()

gmtbegin("Figure6_jl", fmt="pdf,png")
    C = makecpt("-Cthermal.cpt -T0/4/0.1 -Z > rbow.cpt")
    subplot(grid="2x3", panels_size=(8,4), margins="3p") # `autolabel=true or 1` do the automatic labeling
        
        # Figure (a)
        data1 = grdedit("alpha_1.grd", region=range_xy) # change cooridinates scale
        basemap!(region=range_xy, figsize=(8, -4), frame=(axes=:WStr,),
        xaxis=(annot=2, ticks=1), 
        yaxis=(annot=1, label=:"Z-Depth [km]"),
        par=(MAP_FRAME_PEN="1p,black",),panel=(1,1))
        grdimage!(data1, region=range_xy, figsize=(8, -4), cmap="rbow.cpt", panel=(1,1)) 
        text!(["(a) @~\141@~=3.0"], font=12, x=-0.05, y=5.5)

        # Figure (b)
        data2 = grdedit("alpha_2.grd", region=range_xy) # change cooridinates scale
        basemap!(region=range_xy, figsize=(8, -4), frame=(axes=:WStr,),
        xaxis=(annot=2, ticks=1), yaxis=(annot=1),
        par=(MAP_FRAME_PEN="1p,black",), panel=(1,2)) 
        grdimage!(data2, region=range_xy, figsize=(8, -4), cmap="rbow.cpt", panel=(1,2)) 
        text!(["(b) @~\141@~=4.0"], font=12, x=-0.05, y=5.5)

        # Figure (c)
        data3 = grdedit("alpha_3.grd", region=range_xy) # change cooridinates scale
        basemap!(region=range_xy, figsize=(8, -4), frame=(axes=:WStr,),
        xaxis=(annot=2, ticks=1), yaxis=(annot=1),
        par=(MAP_FRAME_PEN="1p,black",), panel=(1,3)) 
        grdimage!(data3, region=range_xy, figsize=(8, -4), cmap="rbow.cpt", panel=(1,3)) 
        text!(["(c) @~\141@~=5.0"], font=12, x=-0.05, y=5.5)

        # Figure (d)
        data4 = grdedit("alpha_4.grd", region=range_xy) # change cooridinates scale
        basemap!(region=range_xy, figsize=(8, -4), frame=(axes=:WStr,),
        xaxis=(annot=2, ticks=1, label=:"Y-Distance [km]"), 
        yaxis=(annot=1, label=:"Z-Depth [km]"),
        par=(MAP_FRAME_PEN="1p,black",), panel=(2,1)) 
        grdimage!(data4, region=range_xy, figsize=(8, -4), cmap="rbow.cpt", panel=(2,1)) 
        text!(["(d) @~\141@~=6.0"], font=12, x=-0.05, y=5.5)

        # Figure (e)
        data5 = grdedit("alpha_5.grd", region=range_xy) # change cooridinates scale
        basemap!(region=range_xy, figsize=(8, -4), frame=(axes=:WStr,),
        xaxis=(annot=2, ticks=1, label=:"Y-Distance [km]"), 
        yaxis=(annot=1),
        par=(MAP_FRAME_PEN="1p,black",), panel=(2,2)) 
        grdimage!(data5, region=range_xy, figsize=(8, -4), cmap="rbow.cpt", panel=(2,2)) 
        text!(["(e) @~\141@~=7.0"], font=12, x=-0.05, y=5.5)
        
        colorbar!(xaxis=(annot=1, ticks=0.2), pos=(paper=true, anchor=(1,2), size=(6.2,0.5), horizontal=true, triangles=true), par=(MAP_FRAME_PEN="1p,black",), panel=(2,3))
        text!(["Log@-10@-[@~W\267@~m]"], font=12, x=0, y=4.3)

    subplot()
gmtend()

File: Plotting_Scripts/Figure6/Figure6.py
import pygmt

# Set file and directory paths
dir_path = "./"
cpt_file_in = f"{dir_path}thermal.cpt"
cpt_file_out = f"{dir_path}rbow.cpt"
alpha_grids = [f"{dir_path}alpha_{i}_1.grd" for i in range(1, 6)]  
out_pdf_png = f"{dir_path}Figure6_py"

# set initial range
xmin, xmax, ymin, ymax, scale = -6400, 6400, 0, 6000, 1000
xmin, xmax = xmin / scale, xmax / scale
ymin, ymax = ymin / scale, ymax / scale
range_xy = f"{xmin}/{xmax}/{ymin}/{ymax}"

# Create a color palette
pygmt.makecpt(cmap=cpt_file_in, series="0/4.0/0.1", continuous=True, output=cpt_file_out)

# Convert grid range using grdedit in original GMT, currently, PyGMT does not support the `grdedit` and `grdconvert` modules
# gmt grdedit alpha_{i}.grd -R-6.4/6.4/0/6 -Galpha_{i}_1.grd

# Create a figure with 2 rows and 3 columns subplots
fig = pygmt.Figure()
pygmt.config(MAP_FRAME_PEN="0.5p")
with fig.subplot(nrows=2, ncols=3, figsize=("24c", "8c"), frame="WSne", margins=["0.8c", "0.5c"]):
    labels = ['(a) @~\141@~=3.0', '(b) @~\141@~=4.0', '(c) @~\141@~=5.0', '(d) @~\141@~=6.0', '(e) @~\141@~=7.0']
    for i, alpha_grd in enumerate(alpha_grids):
        with fig.set_panel(panel=i):
            # adjust frame labels for each subplot
            if i == 0:
                frame = ["xa2f1", "xa2f1", "ya1f1+lZ-Depth (km)", "WS"]
                fig.grdimage(grid=alpha_grd, region=range_xy, projection="X8c/-4c", cmap=cpt_file_out, frame=frame)
            elif i == 3:
                frame = ["xa2f1+lY-Distance (km)", "ya1f1+lZ-Depth (km)", "WS"]
                fig.grdimage(grid=alpha_grd, region=range_xy, projection="X8c/-4c", cmap=cpt_file_out, frame=frame)
            elif i == 4:
                frame = ["xa2f1+lY-Distance (km)", "ya1f1", "WS"]
                fig.grdimage(grid=alpha_grd, region=range_xy, projection="X8c/-4c", cmap=cpt_file_out, frame=frame)
            else:
                frame = ["xa2f1", "ya1f1", "WS"]
                fig.grdimage(grid=alpha_grd, region=range_xy, projection="X8c/-4c", cmap=cpt_file_out, frame=frame)
                
            fig.basemap(region=range_xy, projection="X8c/-4c", frame="ne")
            fig.text(x=-0.5, y=5.0, text=f"{labels[i]}", font="12p,Helvetica", justify="TL")

# add colorbar
fig.colorbar(cmap=cpt_file_out, position="jTC+w6c/0.5c+o9.5c/6.0c+ml+e", frame=["xaf+lLog@-10@-(@~W\\267@~m)", "y"])

# Save the figure to a PDF and PNG file
fig.savefig(f"{out_pdf_png}.png")
fig.savefig(f"{out_pdf_png}.pdf")